src/nomad_simulations/schema_packages/model_method.py

```diff
@@ -81,308 +87,574 @@ class ModelMethod(BaseModelMethod):
+        number of explicitly treated electrons and to incorporate scalar
+        relativistic effects.
+        - **PP**: A pseudopotential is used (common in solid-state or plane-wave
+        codes) to represent the effective interaction of valence electrons with
+        the ionic cores, often norm-conserving or ultrasoft types.
+
+        Both ECPs and PPs reduce the computational cost and can account for
+        relativistic effects in the core region. They differ mainly in historical
+        context and implementation (quantum chemistry vs. solid-state methods).
         """,
     )
 
-    # ? This could be moved under `contributions`, @ndaelman-hu
-    xc_functionals = SubSection(sub_section=XCFunctional.m_def, repeats=True)
+    dkh_order = Quantity(
+        type=np.int32,
+        description='Order used for DKH (e.g., 2, 4, ...); None if not applicable.',
+    )
 
-    exact_exchange_mixing_factor = Quantity(
-        type=np.float64,
+    is_soc_included = Quantity(
+        type=bool,
         description="""
-        Amount of exact exchange mixed in with the XC functional (value range = [0, 1]).
+        Whether the Hamiltonian used **during SCF** already included spin-orbit coupling.
+        If False, spin-orbit may still be added perturbatively later (e.g. SOMF).
         """,
     )
 
-    # ! MEnum this
-    self_interaction_correction_method = Quantity(
-        type=str,
+    soc_treatment = Quantity(
+        type=MEnum('none', 'perturbative', 'second-variational', 'variational'),
+        default='none',
         description="""
-        Contains the name for the self-interaction correction (SIC) treatment used to
-        calculate the final energy and related quantities. If skipped or empty, no special
-        correction is applied.
+        SOC inclusion mode:
+          • none: no SOC anywhere
+          • perturbative: added after scalar SCF (e.g., SOMF, BP, RASSI)
+          • second-variational: diagonalized after scalar SCF (e.g., LAPW)
+          • variational: included self-consistently (2c/4c or PP/PAW noncollinear)
+        """,
+    )
 
-        The following SIC methods are available:
 
-        | SIC method                | Description                       |
+class ModelMethodElectronic(ModelMethod):
+    """
+    A base section used to define the parameters of a model Hamiltonian used in electronic structure
+    calculations (TB, DFT, GW, BSE, DMFT, etc).
+    """
 
-        | ------------------------- | --------------------------------  |
+    # instead of the legacy quantity is_spin_polarized
+    # TODO : this part should be revisited once Spin is handled.
+    reference_determinant = Quantity(
+        type=MEnum('UHF', 'RHF', 'ROHF', 'UKS', 'RKS', 'ROKS'),
+        description="""
+        The type of reference determinant.
+          - UHF / RHF / ROHF : common for wavefunction-based reference
+          - UKS / RKS / ROKS : if a KS-DFT reference is used (rare but possible).        
```

Comment:
I don't think that a schema should comment on how common an approach is. It should just aim to cover as many as possible.

```suggestion
        The type of reference determinant.
          - UHF / RHF / ROHF : Hartree-Fock orbitals
          - UKS / RKS / ROKS : Kohn-Sham orbitals        
```

There are also orbital-free DFs (this is were the kinetic functionals come in), so I think it's better to address the nature of the orbitals constituting the determinants.